tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,297 @@
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, errors.New("failed to get nginxGateway")
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyNginxGatewayConditions := func(ng ngfAPI.NginxGateway) error {
+		if ng.Status.Conditions == nil {
+			return errors.New("nginxGateway is has no conditions")
+		}
+
+		if len(ng.Status.Conditions) != 1 {
+			return fmt.Errorf(
+				"expected nginxGateway to have only one condition, instead has %d conditions",
+				len(ng.Status.Conditions),
+			)
+		}
+
+		return nil
+	}
+
+	getNginxGatewayCurrentObservedGeneration := func(ng ngfAPI.NginxGateway) (int64, error) {
+		if err := verifyNginxGatewayConditions(ng); err != nil {
+			return 0, err
+		}
+
+		return ng.Status.Conditions[0].ObservedGeneration, nil
+	}
+
+	verifyNginxGatewayStatus := func(ng ngfAPI.NginxGateway, expObservedGen int64) error {
+		if err := verifyNginxGatewayConditions(ng); err != nil {
+			return err
+		}
+
+		condition := ng.Status.Conditions[0]
+
+		if condition.Type != "Valid" {
+			return fmt.Errorf(
+				"expected nginxGateway condition type to be Valid, instead has type %s",
+				condition.Type,
+			)
+		}
+
+		if condition.Reason != "Valid" {
+			return fmt.Errorf("expected nginxGateway reason to be Valid, instead is %s", condition.Reason)
+		}
+
+		if condition.ObservedGeneration != expObservedGen {
+			return fmt.Errorf(
+				"expected nginxGateway observed generation to be %d, instead is %d",
+				expObservedGen,
+				condition.ObservedGeneration,
+			)
+		}
+
+		return nil
+	}
+
+	getNGFPodName := func() (string, error) {
+		podNames, err := framework.GetReadyNGFPodNames(
+			k8sClient,
+			ngfNamespace,
+			releaseName,
+			timeoutConfig.GetTimeout,
+		)
+		if err != nil {
+			return "", err
+		}
+
+		if len(podNames) != 1 {
+			return "", fmt.Errorf("expected 1 pod name, got %d", len(podNames))
+		}
+
+		return podNames[0], nil
+	}
+
+	AfterAll(func() {
+		// re-apply NginxGateway crd to restore NGF instance for following functional tests
+		Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+		Eventually(
+			func() bool {
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				if err != nil {
+					return false
+				}
+
+				return verifyNginxGatewayStatus(ng, int64(1)) != nil
```

Comment:
So if the error is nil, we return false here... is that what we want?

Reply:
no... thanks for catching that. I messed this up in the refactor thanks for catching this.